### DISCUSSION TITLE
Filtering by columns of a computed relationship

### DISCUSSION THREAD

**Discussion by newageoflight** [supabase/supabase#28492] [Questions] [answered]
I have a computed relationship that is embedded into my PostgREST query like this:

```typescript
  const { data, error } = await supabase
    .from('protoevents')
    .select('*, events (*), tags (label), related_direct (id)')
    .eq('subtype', 'event')
    .eq('related_direct.id', params.iid)
    .eq('pt_id', params.id);
```

However, I'm finding that filtering by `related_direct.id` is not working at all and the results returned are as though the filter was never applied.

Here is the PL/pgSQL function used to enable that for comparison; I'm also finding that the type generator doesn't automatically generate the types for this so I'm wondering if that has something to do with it.

```plpgsql
CREATE OR REPLACE FUNCTION related_protoevents_direct(protoevents, protoevent_relation, reverse_direction BOOLEAN DEFAULT FALSE) RETURNS SETOF protoevents AS $$
  SELECT pe.*
  FROM public.protoevents pe
  JOIN public.protoevents_edges pee ON pe.id = (CASE WHEN reverse_direction THEN pee.peid2 ELSE pee.peid1 END)
  WHERE (CASE WHEN reverse_direction THEN pee.peid1 ELSE pee.peid2 END) = $1.id AND pee.label = $2
$$ STABLE LANGUAGE SQL;

...

CREATE OR REPLACE FUNCTION related_direct(protoevents) RETURNS SETOF protoevents AS $$ SELECT related_protoevents_direct($1, 'related'); $$ STABLE LANGUAGE SQL;
```

Any way to fix this?

**Comment by newageoflight** [comment]
Since opening this I've noted there's a bug open on `@supabase/postgrest-js` on the same subject: https://github.com/supabase/postgrest-js/issues/474; on inspection of my `node_modules` folder the version of `postgrest-js` that came with Supabase was 1.15.8 and the current version of Supabase is 2.45.0

EDIT: Never mind, downgrading Supabase doesn't do anything

EDIT 2: Also noting that trying to access `item.related_direct.id` when looping over `data` raises a `SelectQueryError<"Referencing missing column 'id'">[]`

**Comment by gc-ft** [comment, marked as answer]
Try

```
  const { data, error } = await supabase
    .from('protoevents')
    .select('*, events (*), tags (label), related_direct!inner(id)')
    .eq('subtype', 'event')
    .eq('related_direct.id', params.iid)
    .eq('pt_id', params.id);
```

**Reply by newageoflight** [reply]
Works, thanks. How on Earth did I miss this 🤣
